### DISCUSSION TITLE
Advanced optimus

### DISCUSSION THREAD

**Discussion by barci2** [NVIDIA/open-gpu-kernel-modules#430] [Q&A]
Does the driver currently have support for using advanced optimus? Is there any work being done on it? What is the best management software for utilizing the capability user-side?

**Comment by aritger** [comment]
As of the 525.xx open-gpu-kernel-modules releases, power management support is still not ready.  You are probably better off using the proprietary NVIDIA driver for now until power management support is ready in open-gpu-kernel-modules (it is actively being worked on, but I don't have a firm ETA).

In terms of how to configure, please see these chapters from the NVIDIA Linux GPU driver README:

http://us.download.nvidia.com/XFree86/Linux-x86_64/525.60.11/README/optimus.html
http://us.download.nvidia.com/XFree86/Linux-x86_64/525.60.11/README/randr14.html
http://us.download.nvidia.com/XFree86/Linux-x86_64/525.60.11/README/primerenderoffload.html

**Reply by barci2** [reply]
do you know then whether the proprietary driver has the support for dynamic mux switching, or where/who to ask about that?

**Reply by aritger** [reply]
Yes, I believe the proprietary driver handles dynamic mux switching today, though I'm not too familiar with the details of it myself.  I'll try to ask internally and reply here (it may not be until after the holidays).

**Reply by barci2** [reply]
Alright, thanks. Important thing to note here is whether it's possible to switch the mux **dynamically** with advanced optimus, i.e. when the os is still running

**Reply by wassou93** [reply]
Hi @aritger,

Do you have any updates on this?
I have a Legion 5i Pro 2022 laptop,
I'm using Ubuntu 23.10 with 535.129.03 driver, is there any chance I can enable automatic mux switching like in windows 11 ?

```

wassou@was-pc:~$ nvidia-smi 
Tue Nov 14 10:30:00 2023       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3060 ...    Off | 00000000:01:00.0  On |                  N/A |
| N/A   37C    P8              11W /  80W |     44MiB /  6144MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      1311      G   /usr/lib/xorg/Xorg                           24MiB |
+---------------------------------------------------------------------------------------+
wassou@was-pc:~$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 23.10"
NAME="Ubuntu"
VERSION_ID="23.10"
VERSION="23.10 (Mantic Minotaur)"
VERSION_CODENAME=mantic
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=mantic
LOGO=ubuntu-logo

```

**Reply by aritger** [reply]
Sorry, we don't currently support automatic mux switching on Linux.

**Comment by gregory112** [comment]
What about this thread? https://lists.freedesktop.org/archives/dri-devel/2022-November/379107.html

**Comment by 500InternalError** [comment]
Is there any update?

**Comment by ltsdw** [comment]
Sorry to bump this for you guys who are subscribed 😔 

But from the current discussion I'm still confused whether it's not supported on the open-gpu-kernel-modules, but it's supported on the proprietary, or is it not supported at all?

If the later, shouldn't we be discussing it more actively? It was "hyped" in 2022, we're now heading towards 2026.
